src/main/java/codeu/model/store/basic/MessageStore.java

```diff
@@ -114,8 +138,27 @@ public void addMessage(Message message) {
   /** Sets the List of Messages stored by this MessageStore. */
   public void setMessages(List<Message> messages) {
     this.messages = messages;
+    //messagesMap = new HashMap<UUID, ArrayList<Message>>();
+    ArrayList<Message> children;;
+    for(Message message : messages){
+      if(message.getParentId() != null){
+        if(messagesMap.containsKey(message.getParentId())){
+          children = messagesMap.get(message.getParentId());
+        }else{
+          children = new ArrayList<Message>();
+        }
+        children.add(message);
+        messagesMap.replace(message.getParentId(), children);
```

Comment:
I think this is not needed because the children array is a pointer. Hence, when we add the the message to the array, the copy in the hashmap should be updated because they have the same reference.